### PR TITLE
Service Principal auth

### PR DESCRIPTION
**Pull request by KonstantinTyukalov**
This PR adds `SP` auth approach.
To do so, you'll need to set:
- `auth`: `SP`
- `clientid`: `<your client(app) id>`
- `tenantid`: `<your tenant id>`
- `clientsecret`: `<your client secret value>`

Work item: internal 2032949

### CODE REVIEW

**Review by ismayilov-ismayil — APPROVED** [review]

**Review by LiliaSabitova — APPROVED** [review]

**Review by tkasparek — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

File: src/Agent.Listener/Configuration/CredentialManager.cs [hunk 1]
```diff
@@ -26,6 +26,7 @@ public class CredentialManager : AgentService, ICredentialManager
             { Constants.Configuration.Integrated, typeof(IntegratedCredential)},
             { Constants.Configuration.OAuth, typeof(OAuthCredential)},
             { Constants.Configuration.ServiceIdentity, typeof(ServiceIdentityCredential)},
+            { Constants.Configuration.ServicePrincipal, typeof(ServicePrincipalCredential)},
```
[the remark this answers is not in the document]
Comment (KonstantinTyukalov) [reply src/Agent.Listener/Configuration/CredentialManager.cs:29]: Will be done in separate PR

**Comment by akanieski** [comment]
It was mentioned elsewhere about managed identity support, what about client certificate support for the service principal? Many consider client secrets as less ideal (as is suggested by the AAD best practices docs [here](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/service-accounts-principal) . I know the agent currently supports client certs, can we just piggy back on those existing configuration options?

**Comment by avekrivoy** [comment]
When will this feature be available?

**Comment by KonstantinTyukalov** [comment]
Blocked while telemetry changes are not merged: https://github.com/microsoft/azure-pipelines-agent/pull/4333

File: src/Agent.Listener/Configuration/CredentialManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (KonstantinTyukalov) [reply src/Agent.Listener/Configuration/CredentialManager.cs:29]: done here: https://github.com/microsoft/azure-pipelines-agent/pull/4333

**Comment by avekrivoy** [comment]
> Blocked while telemetry changes are not merged: #4333

I see it's already merged. Any ETA on this? Really waiting for this SP auth feature because I want to get rid of PATs in my agent's configuration

**Review by kirill-ivlev — APPROVED** [review]

**Pull request merged**